server/service/transport_test.go

```diff
@@ -82,35 +84,373 @@ func TestListOptionsFromRequest(t *testing.T) {
 			shouldErr400: true,
 		},
 		{
-			url:          "/foo?page=-1&per_page=-10",
+			url:          "/foo?page=1&per_page=-10",
 			shouldErr400: true,
 		},
+		// order_direction without order_key
 		{
 			url:          "/foo?page=1&order_direction=desc",
 			shouldErr400: true,
 		},
+		// bad order_direction
 		{
-			url:          "/foo?&order_direction=foo&order_key=",
+			url:          "/foo?&order_direction=foo&order_key=foo",
```

Comment:
These old tests weren't doing the right thing. Fixed them because I saw holes in coverage.